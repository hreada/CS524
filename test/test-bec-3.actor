let(
  bind(append, 
       rec(lambda(beh, lambda(current, lambda(new,
         become(app(beh, pr(new, current)))))))),
  letactor(
    bind(a, app(append, nil)),
    seq(send(a, 0),
    seq(send(a, 1),
    seq(send(a, 2),
        send(a, 3))))))
  
