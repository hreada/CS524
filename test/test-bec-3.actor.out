<T>
  <actors>
    <actor>
      <k> exec ( nil ) </k>
      <id> _41 </id>
    </actor>
    <actor>
      <k> ready ( lambda ( new , become ( app ( lambda ( _2 , app ( app ( lambda ( _1 , app ( lambda ( beh , lambda ( current , lambda ( new , become ( app ( beh , pr ( new , current ) ) ) ) ) ) , lambda ( _2 , app ( app ( _1 , _1 ) , _2 ) ) ) ) , lambda ( _1 , app ( lambda ( beh , lambda ( current , lambda ( new , become ( app ( beh , pr ( new , current ) ) ) ) ) ) , lambda ( _2 , app ( app ( _1 , _1 ) , _2 ) ) ) ) ) , _2 ) ) , pr ( new , pr ( 3 , pr ( 2 , pr ( 1 , pr ( 0 , nil ) ) ) ) ) ) ) ) ) </k>
      <id> _3 </id>
    </actor>
    <actor>
      <k> exec ( nil ) </k>
      <id> initactor </id>
    </actor>
    <actor>
      <k> exec ( nil ) </k>
      <id> _16 </id>
    </actor>
    <actor>
      <k> exec ( nil ) </k>
      <id> _25 </id>
    </actor>
    <actor>
      <k> exec ( nil ) </k>
      <id> _34 </id>
    </actor>
  </actors>
  <messages> .MessageCellBag </messages>
  <definedaddr> ListItem ( initactor ) ListItem ( _3 ) ListItem ( _16 ) ListItem ( _25 ) ListItem ( _34 ) ListItem ( _41 ) </definedaddr>
</T>
